### PR TITLE
Fix shellcheck issues on shell scripts

### PR DESCRIPTION
**Pull request by nightlark**
### Summary
If merged this pull request will fix various shellcheck warnings and errors. Mostly quoting variables to make sure spaces in the value work correctly, replacing `which` with the more standard `command -v` to get program paths, and adding `|| exit/return` for failed directory changes.

### Proposed changes
- Fix shellcheck warnings, including some bugs
- Remove no longer used Boost sanitizer and compilation step; copy the Boost headers instead (160 seconds -> 50 seconds for installing dependencies without a cache on Travis)

### CODE REVIEW

**Comment by phlptp** [comment]
Why are the Azure builds still failing?

**Comment by nightlark** [comment]
Probably because the build was for the branch (which was created before the Python 3.4 change got merged into develop) and not the PR. I rebased on develop to get the updated Azure yml file into this branch and now it's using the right version.

File: .github/workflows/interface_gen_script.sh [hunk 1]
```diff
@@ -1,16 +1,16 @@
-#!/usr/bin/sh
+#!/usr/bin/bash
 
 # Generate Matlab interface using swig matlab
 mkdir build_matlab
-pushd build_matlab
+pushd build_matlab || exit
```
Comment (phlptp) [inline .github/workflows/interface_gen_script.sh]: what does the ||exit do exactly.  I would guess it just just forces the return result of this command to be 0?

**Comment by phlptp** [comment]
are all these scripts running somewhere?

File: .github/workflows/interface_gen_script.sh (hunk 1 above)
Comment (nightlark) [reply .github/workflows/interface_gen_script.sh]: The default exit status is 0. The main thing is that if changing directories fails it stops running the rest of the script.

**Comment by nightlark** [comment]
Only one of the git archive related scripts runs as part of the release/nightly build action.

mktempdir.sh is a mystery, but it looks like I committed it for some unknown reason back in May.

The rest all run as part of various CI builds.

**Comment by phlptp** [comment]
Is this ready to merge?

**Comment by nightlark** [comment]
I think it's okay to merge. I got rid of mktempdir.sh since it isn't used anywhere, I think I'd used it to test making a temporary directory in a portable way (macOS mktemp command has quirks that Linux doesn't have).

**Pull request merged**